benchmarks/envoy_proxy.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+"""@package envoy_proxy.py
+
+Contains customized fixture & EnvoyProxyServer abstraction for use in tests.
+"""
+
+import logging
+import os
+import pytest
+
+from test.integration.integration_test_fixtures import (HttpIntegrationTestBase,
+                                                        determineIpVersionsFromEnvironment)
+from test.integration.nighthawk_test_server import NighthawkTestServer
+
+
+class EnvoyProxyServer(NighthawkTestServer):
+  """
+  Envoy proxy server abstraction. Note that it derives from NighthawkTestServer, as that
+  is implemented as a customized Envoy, which is convenient here: the CLI and admin interface
+  mechanics that we rely on are the same. So all we do here, is specialize so we can override
+  the docker image and binary name.
+  """
+
+  def __init__(self, config_template_path, server_ip, ip_version, parameters=dict(), tag=""):
+    # If no explicit envoy path is passed, we'll use nighthawk_test_server.
+    super(EnvoyProxyServer, self).__init__(
+        os.getenv("ENVOY_PATH", "nighthawk_test_server"),
+        config_template_path,
+        server_ip,
+        ip_version,
+        parameters=parameters,
+        tag=tag)
+    self.docker_image = os.getenv("ENVOY_DOCKER_IMAGE_TO_TEST", "")
+
+
+@pytest.fixture()
+def proxy_config():
+  yield "nighthawk/benchmarks/configurations/envoy_proxy.yaml"
+
+
+class InjectHttpProxyIntegrationTestBase(HttpIntegrationTestBase):
+  """
+  Fixture which spins up a Nighthawk test server which listens on plain http,
+  and set up an Envoy instances to proxy to that, als listening on plain http.
+  """
+
+  def __init__(self, ip_version, server_config, proxy_config):
+    """See base class."""
+    super(InjectHttpProxyIntegrationTestBase, self).__init__(ip_version, server_config)
+    self.proxy_config = proxy_config
+
+  def setUp(self):
+    super(InjectHttpProxyIntegrationTestBase, self).setUp()
+    logging.info("injecting envoy proxy ...")
+    # TODO(oschaaf): how should this interact with multiple backends?
```

Comment:
Great point; added this to the TODO list in `README.md`. This may or may not be a doc issue, possibly this can already be made to work with the status quo and just needs a clear example of how to do it. (The NH integration tests does have tests involving multiple origins).